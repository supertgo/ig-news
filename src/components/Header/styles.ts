import styled, { css } from 'styled-components';

export const Wrapper = styled.header`
  height: 8rem;
  border-bottom: 1px solid #29292e;
`;

export const Content = styled.div`
  ${({ theme }) => css`
    max-width: 112rem;
    height: 8rem;
    margin: 0 auto;
    padding: 0 ${theme.spacings.medium};

    display: flex;
    align-items: center;

    button {
      margin-left: auto;
    }
  `}
`;

export const Image = styled.img``;

export const Nav = styled.nav`
  margin-left: 8rem;
  height: 8rem;
`;

type LinkProps = {
  isActive?: boolean;
};

export const Link = styled.a<LinkProps>`
  ${({ theme, isActive }) => css`
    display: inline-block;
    position: relative;
    padding: 0 ${theme.spacings.xxsmall};
    height: 8rem;
    line-height: 8rem;
    color: ${theme.colors.text};
    transition: color 0.2s;
    font-size: ${theme.font.sizes.medium};

    & + a {
      margin-left: ${theme.spacings.medium};
    }

    &:hover {
      color: ${theme.colors.white};
    }

    ${isActive &&
    css`
      & {
        color: ${theme.colors.white};
        font-weight: ${theme.font.bold};
      }

      ::after {
        content: '';
        height: 3px;
        border-radius: 3px 3px 0 0;
        width: 100%;
        position: absolute;
        bottom: 0;
        left: 0;
        background: ${theme.colors.yellow};
      }
    `}
  `}
`;
