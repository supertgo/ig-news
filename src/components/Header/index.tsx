import Link from 'next/link';
import { useRouter } from 'next/router';

import SignInButton from 'components/SignInButton';

import * as S from './styles';

const Header = () => {
  const { asPath } = useRouter();
  return (
    <S.Wrapper>
      <S.Content>
        <S.Image src="/images/logo.svg" alt="ig.news" />
        <S.Nav>
          <Link href="/" passHref>
            <S.Link isActive={asPath === '/'}>Home</S.Link>
          </Link>
          <Link href="/posts" passHref>
            <S.Link isActive={asPath === '/posts'}>Posts</S.Link>
          </Link>
        </S.Nav>

        <SignInButton />
      </S.Content>
    </S.Wrapper>
  );
};

export default Header;
