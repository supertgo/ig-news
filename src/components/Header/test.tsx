import { screen } from '@testing-library/react';
import { renderWithTheme } from 'utils/tests/helpers';

import Header from '.';

jest.mock('components/SignInButton', () => {
  return {
    __esModule: true,
    default: function Mock() {
      return <div data-testid="Mock SignInButton"></div>;
    }
  };
});

jest.mock('components/ActiveLink', () => {
  return {
    __esModule: true,
    default: function Mock() {
      return <div data-testid="Mock ActiveLink"></div>;
    }
  };
});

describe('<Header />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(<Header />);

    expect(screen.getByRole('img', { name: /ig\.news/i })).toBeInTheDocument();
    expect(screen.getByTestId('Mock SignInButton')).toBeInTheDocument();
    expect(screen.getAllByTestId('Mock ActiveLink')).toHaveLength(2);

    expect(container.firstChild).toMatchSnapshot();
  });
});
