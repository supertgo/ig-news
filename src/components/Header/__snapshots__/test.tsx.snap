// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render the heading 1`] = `
.c0 {
  height: 8rem;
  border-bottom: 1px solid #29292e;
}

.c1 {
  max-width: 112rem;
  height: 8rem;
  margin: 0 auto;
  padding: 0 3.2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 button {
  margin-left: auto;
}

.c2 {
  margin-left: 8rem;
  height: 8rem;
}

.c3 {
  display: inline-block;
  position: relative;
  padding: 0 0.8rem;
  height: 8rem;
  line-height: 8rem;
  color: #A8A8B3;
  -webkit-transition: color 0.2s;
  transition: color 0.2s;
  font-size: 1.6rem;
}

.c3 + a {
  margin-left: 3.2rem;
}

.c3:hover {
  color: #FFFFFF;
}

<header
  class="c0"
>
  <div
    class="c1"
  >
    <img
      alt="ig.news"
      class=""
      src="/images/logo.svg"
    />
    <nav
      class="c2"
    >
      <a
        class="c3"
        href="/"
      >
        Home
      </a>
      <a
        class="c3"
        href="/posts"
      >
        Posts
      </a>
    </nav>
    <div
      data-testid="Mock SignInButton"
    />
  </div>
</header>
`;
